### PR TITLE
Sync development

### PR DESCRIPTION
**Pull request by gmazzap**
- Add support for custom env config files in 'vip-composer-plugin-env-config' packages, see #14 
- Upgraded dev dependencies and fix newly found issues
- Modernize actions used in worfklows and run tests for PHP 8.3
- Remove license header from each file

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes/features)
- [x] Docs have been added/updated (for bug fixes/features)

**What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

docs update, deps update

**What is the current behavior?** (You can also link to an open issue here)

n/a

**What is the new behavior (if this is a feature change)?**

See #14 

**Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

no

**Other information**:

### CODE REVIEW

File: .github/workflows/quality-assurance-php.yml [hunk 1]
```diff
@@ -35,7 +35,7 @@ jobs:
         if: ${{ (github.event_name != 'workflow_dispatch') || ((github.event.inputs.jobs == 'Run all') || (github.event.inputs.jobs == 'Run lint only')) }}
         strategy:
             matrix:
-                php: [ "8.0", "8.1", "8.2" ]
+                php: [ "8.0", "8.1", "8.2", "8.3" ]
```
Comment (Chrico) [inline .github/workflows/quality-assurance-php.yml:38]: WordPress VIP does not support PHP 8.0 anymore. Min version in their background is PHP 8.1 for configuration.

**Review by Chrico — COMMENTED** [review]
looks good to me so far. Just 2 smaller comments. 👍🏻

File: .github/workflows/quality-assurance-php.yml (hunk 1 above)
Comment (gmazzap) [reply .github/workflows/quality-assurance-php.yml:38]: They don't support for new sites, but existing sites might still bee on 8.0. We have a few sites still on 8.0 right now.

File: src/Config.php [hunk 2]
```diff
@@ -143,47 +137,73 @@ public function composerLockPath(): string
...
+    }
+
+    /**
+     * @return list<non-empty-string>
+     */
+    public function envConfigs(): array
+    {
+        $customEnvs = (array) $this->offsetGet(self::CUSTOM_ENV_NAMES_KEY);
+        if ($customEnvs === []) {
+            return ['local', 'development', 'staging', 'production', 'all'];
```
[the remark this answers is not in the document]
Comment (gmazzap) [reply src/Config.php]: Done

**Pull request merged**